test/System.Web.Http.Owin.Test/HttpMessageHandlerAdapterTest.cs

```diff
@@ -429,7 +429,7 @@ public async Task Invoke_BuildsUriWithHostAndPort()
         [InlineData(@"-_.~+""<>^`{|}")]
         // random unicode characters
         [InlineData("激光這")]
-        [InlineData("%24")]
+        [InlineData("%24", Skip="Decoded to '$' at the moment. !!! Investigate !!!")]
```

Comment:
Can we link an issue here?

Reply:
@Tratcher is this related to a known breaking change in the underlying Owin packages❔

Reply:
I don't think so. What versions did you update from and to?

'$' is not a reserved character in request paths, it's supposed to be un-escaped.

Reply:
We went from Microsoft.Owin v2.0.2 to v4.2.2

Reply:
Maybe https://github.com/aspnet/AspNetKatana/commit/7db01eb2db3acf787651b4dc27d5fd05dc47e4ec, but that was 4.0.0, years ago.

Reply:
That makes sense given how aged our dependency was. I'll create a `Invoke_CreatesUri_ContainingCorrectlyDecodedStrings(string encoded, string decoded)` alternative to this test. Will include `[InlineData("%24", "$")]`. Do you have a suggestion or two for additional encodings that should be decoded when presented to the application❔

Reply:
pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
   sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
                 / "*" / "+" / "," / ";" / "="

Reply:
Yeah, I saw that BNF and realized the comments w/ the `[TheoryData(...)]` above are pretty off-base. Decided to just test percent encodings of `$`, `(`, `)`, `[`, `]`, `{`, and `}` rather than recategorizing things here.